src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -141,15 +145,24 @@ private static String retrieveText(ByteString imageBytes)
             "Received image annotation response without text annotations.");
       }
 
-      // First element has the entire raw text from the image
-      EntityAnnotation annotation = response.getTextAnnotationsList().get(0);
+      // First element has the entire raw text from the image.
+      EntityAnnotation textAnnotation = response.getTextAnnotationsList().get(0);
 
-      rawText = annotation.getDescription();
+      String rawText = textAnnotation.getDescription();
+      analysisBuilder = new AnalysisResults.Builder(rawText);
+
+      // If a logo was detected with a confidence above the threshold, use it to set the store.
```

Comment:
I'm guessing the threshold is out of 1? And the closer to 1 you are the more accurate the result? Maybe add a comment so this is clear and no assumptions need to be made